### PR TITLE
Fix/clang compiler errors

### PR DESCRIPTION
**Pull request by HiddenBug**
Fix clang compiler errors.

### CODE REVIEW

File: prbt_hardware_support/test/unittests/unittest_speed_observer.cpp [hunk 1]
```diff
@@ -148,14 +148,14 @@ MATCHER_P2(SpeedAtIGe, i, x,
...
-  return static_cast<unsigned long>(i) < arg->name.size() && arg->speed.at(i) >= x;
+  return static_cast<unsigned long>(i) < arg->name.size() && arg->speed.at(static_cast<size_t>(i)) >= x;
 }
 
 MATCHER_P2(SpeedAtILe, i, x,
            "Speed at index " + PrintToString(i) + std::string(negation ? "is not" : "is") + " less or equal to" +
            PrintToString(x) + ".")
 {
-  return static_cast<unsigned long>(i) < arg->name.size() && arg->speed.at(i) <= x;
+  return static_cast<unsigned long>(i) < arg->name.size() && arg->speed.at(static_cast<size_t>(i)) <= x;
```
Comment (leunMar) [inline prbt_hardware_support/test/unittests/unittest_speed_observer.cpp]:
```suggestion
  return static_cast<size_t>(i) < arg->name.size() && arg->speed.at(static_cast<size_t>(i)) <= x;
```

File: prbt_hardware_support/test/unittests/unittest_speed_observer.cpp [hunk 2]
```diff
@@ -148,14 +148,14 @@ MATCHER_P2(SpeedAtIGe, i, x,
            "Speed at index " + PrintToString(i) + std::string(negation ? "is not" : "is") + " greater or equal to" +
            PrintToString(x) + ".")
 {
-  return static_cast<unsigned long>(i) < arg->name.size() && arg->speed.at(i) >= x;
+  return static_cast<unsigned long>(i) < arg->name.size() && arg->speed.at(static_cast<size_t>(i)) >= x;
```
Comment (leunMar) [inline prbt_hardware_support/test/unittests/unittest_speed_observer.cpp]:
```suggestion
  return static_cast<size_t>(i) < arg->name.size() && arg->speed.at(static_cast<size_t>(i)) >= x;
```

Comment (leunMar) [reply prbt_hardware_support/test/unittests/unittest_speed_observer.cpp]: This is more consistent and makes it easier to read. Also the Matcher above (Line 144, was not changed, so I cannot review it) has two casts, where the types now should be replaced by `size_t`.

**Review by leunMar — CHANGES_REQUESTED** [review]
I approve this essentially. Only minor improvements.

**Comment by leunMar** [comment]
Please unify size_t usiage here, too:
https://github.com/PilzDE/pilz_robots/blob/f483eaed601fcf5782cf60e24b4c05c42d1440b6/prbt_hardware_support/test/unittests/unittest_speed_observer.cpp#L144

**Comment by HiddenBug** [comment]
> Please unify size_t usiage here, too:
> 
> https://github.com/PilzDE/pilz_robots/blob/f483eaed601fcf5782cf60e24b4c05c42d1440b6/prbt_hardware_support/test/unittests/unittest_speed_observer.cpp#L144

Done.

**Comment by HiddenBug** [comment]
@leunMar What do you think? Can we merge?

**Review by leunMar — APPROVED** [review]

**Pull request merged**